external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriteTask.scala

```diff
@@ -53,7 +52,7 @@ private[kafka010] class KafkaWriteTask(
     if (producer != null) {
       producer.flush()
       checkForErrors()
-      producer = null
+      CachedKafkaProducer.close(producerConfiguration)
```

Comment:
`CachedKafkaProducer.close` "closes" the producer instead of returning it to pool. Current cache pool doesn't have explicit method to return the producer to pool, so that's why we make producer being `null`.